### PR TITLE
Introduce sagas for transactions

### PR DESCRIPTION
**Pull request by bthesorceror**
- Implement for labels createAll

### CODE REVIEW

File: src/wrappers/labels.ts [hunk 1]
```diff
@@ -64,8 +65,20 @@ export default class {
...
+        },
+        rollback: async (r?: ILabel) => {
+          if (r && r.id) {
+            this.delete(r.id)
+          }
+        },
+      }
+    })
+
+    return await saga(pendingLabels)
```
Comment (ebb-tide) [inline src/wrappers/labels.ts]: shouldn't there be a type generic passed in here?

File: src/utils/sagas.ts [hunk 2]
```diff
@@ -0,0 +1,30 @@
+interface Saga<T> {
+  action: () => Promise<T>
+  rollback: (result?: T) => Promise<void>
+  result?: T
+}
+
+async function rollback<T>(index: number, sagas: Saga<T>[]) {
+  for (let i = index; i >= 0; i--) {
+    await sagas[index].rollback(sagas[i].result)
```
Comment (ebb-tide) [inline src/utils/sagas.ts]: typo!

File: src/wrappers/labels.ts (hunk 1 above)
Comment (bthesorceror) [reply src/wrappers/labels.ts]: double checked that it was properly inferred :)

File: src/utils/sagas.ts (hunk 2 above)
Comment (bthesorceror) [reply src/utils/sagas.ts]: fixed!

**Pull request merged**